src/local/room_position/approximate_offsets.rs

```diff
@@ -0,0 +1,147 @@
+//! Methods related to approximating positions between other positions.
+use crate::objects::HasPosition;
+
+use super::Position;
+
+impl Position {
+    /// Calculates an approximate midpoint between this point and the target.
```

Comment:
This is not a midpoint (as in, it's not in the middle). It's `distance_towards_target` from this point towards the other point. (with big caveats)

Reply:
Also, this line is suspiciously like the first one of the next function 😛

Reply:
Maybe: `midpoint` -> `offset`